code/mob/living/silicon/robot.dm

```diff
@@ -2326,7 +2326,12 @@
 
 		var/coords = src.get_coords()
 		var/area/myarea = get_area(src)
-		message = "CONTACT LOST: [src][coords] in [myarea]"
+		if (modifier == 1) //suicide
+			message = "SELF-TERMINATION DETECTED: [src][coords] in [myarea]"
+		else if (modifier == 2) //killswitch
+			message = "KILLSWITCH ACTIVATED: [src][coords] in [myarea]"
+		else //normal death and gib
+			message = "CONTACT LOST: [src][coords] in [myarea]"
```

Comment:
use a #define for this, you can make it local to this file

ex.
```
#define MOD_NONE 0
#define MOD_SUICIDE 1
#define MOD_KILLSWITCH 2
```

Reply:
I've changed the modifier to use define's as requested by Zewaka. 

I placed them in a new file  _stdlib/code/_robot.dm per conversation with Pali. The defines are used in both suicide.dm and robot.dm, but when defined in robot.dm were unavailable in suicide.dm. As such the new  _stdlib/code/_robot.dm resolves this by making them available to both. I also used MOD_BORG_SUICIDE instead of suicide just to clarify that this is a robot suicide and not human or critter.

After speaking with Urs I've change from multiple else if's to a switch statement. Urs tells me this is more efficient, and better practice.